### DISCUSSION TITLE
What are the rules for custom javascript snippets?

### DISCUSSION THREAD

**Discussion by lartpang** [AutomaApp/automa#40] [Q&A]
What are the rules for custom javascript snippets? Hope to provide some more detailed examples.

Because I want to use javascript to download the image of the current page, but I don't know how to set the input.

**Comment by Kholid060** [comment]
You need to use the active tab block if you want to execute your custom javascript code on the current page (like the first image below). And you can execute the workflow from the extension popup when you're on a website (second image). 
![image](https://user-images.githubusercontent.com/22908993/140897186-3719ce6a-b665-4ec5-a3b6-f2b885a2d2e8.png)
![image](https://user-images.githubusercontent.com/22908993/140906134-a21c1fc8-3097-4858-8b21-93adc4018f5c.png)
